app/assets/javascripts/components/accessible-autocomplete.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-env jquery */
+/* global accessibleAutocomplete */
+// = require accessible-autocomplete/dist/accessible-autocomplete.min.js
+
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  function AccessibleAutocomplete ($module) {
+    this.$module = $module
+  }
+
+  AccessibleAutocomplete.prototype.init = function () {
+    this.$selectElem = this.$module.querySelector('select')
+
+    var configOptions = {
+      selectElement: this.$selectElem,
+      showAllValues: true,
+      confirmOnBlur: true,
+      preserveNullOptions: true, // https://github.com/alphagov/accessible-autocomplete#null-options
+      defaultValue: ''
+    }
+
+    configOptions.onConfirm = this.onConfirm.bind(this)
+
+    new accessibleAutocomplete.enhanceSelectElement(configOptions) // eslint-disable-line no-new, new-cap
+  }
+
+  AccessibleAutocomplete.prototype.onConfirm = function (label, value, removeDropDown) {
+    function escapeHTML (str) {
+      return new window.Option(str).innerHTML
+    }
+
+    if (this.$selectElem.dataset.trackCategory !== undefined && this.$selectElem.dataset.trackAction !== undefined) {
+      track(this.$selectElem.dataset.trackCategory, this.$selectElem.dataset.trackAction, label, this.$selectElem.dataset.trackOptions)
+    }
+    // This is to compensate for the fact that the accessible-autocomplete library will not
+    // update the hidden select if the onConfirm function is supplied
+    // https://github.com/alphagov/accessible-autocomplete/issues/322
+    if (typeof label !== 'undefined') {
+      if (typeof value === 'undefined') {
+        value = Array.from(this.$selectElem.querySelectorAll('option'))
+          .find(function (option) { return option.innerHTML === escapeHTML(label) })
+          .value
+      }
+
+      // if removeDropDown we are clearing the selection from outside the component
+      if (typeof value !== 'undefined' && typeof removeDropDown === 'undefined') {
+        this.$selectElem.value = value
+        this.$selectElem.dispatchEvent(new Event('change'))
+      }
+    }
+  }
+
+  function track (category, action, label, options) {
+    if (window.GOVUK.analytics && window.GOVUK.analytics.trackEvent) {
+      options = options || {}
+      options.label = label
+
+      window.GOVUK.analytics.trackEvent(category, action, options)
```

Comment:
I think this is a hangover from the original code - it's calling the Universal Analytics code for Google Analytics tracking (in the components gem). I believe content-data-admin uses GTM for tracking now, so this is probably not needed? I don't know what the tracking requirements for this component are, so it might be out of the scope of this PR to implement it here.

Reply:
If UA tracking is not required I think this entire function and the `if` statement on line 36 can be removed as well.

Reply:
Hey Andy, thanks for the review 🙏 

If we want to learn more about the tracking requirements for this app, who should we speak with? Or should we just remove the tracking from this component entirely and consider it out of scope for this PR?

Reply:
I've done some more digging and it looks like we aren't tracking interactions with this component anywhere in this app:

The tracking is only set up if `data-track-category` and `data-track-action` are both set on the `<select>` element. These attributes would be set in `app/views/components/_accessible_autocomplete.html.erb` if a `data_attributes` local is passed in. I've had a look through all the views where we include this partial, and none of them pass any locals when rendering the partial.

Given this, I'm going to remove the tracking code, which (I think) also allows us to remove the entire `onConfirm` function (as the remainder of the function just seems to be a workaround for a bug caused by the fact that we override `onConfirm`.

I'll try to summarise this in the commit message.

Reply:
Sorry for the slow response, yes, I think removing the tracking code entirely is the right approach. Removing the `onConfirm` also cleans up the code nicely, although can you check that the hidden `select` element is still updated when the user selects something in the autocomplete?